qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,65 @@
+import logging
+
+from virttest import error_context, env_process
+from virttest.utils_virtio_port import VirtioPortTest
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) transfer data from host to guest but do not read in guest
+    5) repeat step 1 to 4 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    @error_context.context_aware
+    def send_data_from_guest_to_host():
+        session = vm.wait_for_login()
+        port.open()
+        error_context.context("send data from guest to host", logging.info)
+        if params['os_type'] == 'windows':
+            vport_name = '\\\\.\\Global\\' + port.name
+            cmd = 'dd if=/dev/zero of=%s bs=1024 count=1' % vport_name
+        else:
+            cmd = 'dd if=/dev/zero of=/dev/virtio-ports/%s bs=1024 count=1' % port.name
+        session.cmd(cmd)
+        session.close()
+
+    @error_context.context_aware
+    def send_data_from_host_to_guest():
+        port.open()
+        error_context.context("send data from host to guest", logging.info)
+        data = 'Hello world \n' * 100
+        data = data.encode()
+        port.sock.send(data)
+        guest_worker.cmd("virt.open('%s')" % port.name)
```

Comment:
From my checking, no need to open port in guest since we do not need to receive data in this test.
This is different with send data from guest to host.

Reply:
if we open this port but not to read the data it is reasonable. does this operation of  `guest_worker.cmd("virt.open('%s')" % port.name)` include reading data?

Reply:
@nanliu-r  It not include reading date, 'virt.recv' is for reading data.
From my checking,  when send data from guest to host, host port must open first, otherwise data can not be send out (host only open, do not read).
For sending data from host to guest, it is not need to open port in guest. But 'open' it is ok too, no side effect

Reply:
ok, let's keep it.